app/models/endpoint.rb

```diff
@@ -41,6 +41,18 @@ def ssl_cert_store
     nil # use system defaults
   end
 
+  # Split concatenated PEM certs to a list
+  def certificate_authorities
+    return [] if certificate_authority.blank?
+    certificate_authority.split(/(?=-----BEGIN)/).reject(&:blank?)
```

Comment:
That is true for the content of the certificate itself, but not for the text that goes before or after it. For example:

```
This is my text, with a -----BEGIN to break you.
-----BEGIN CERTIFICATE-----
The actual certificate
-----END CERTIFICATE-----
```

As I said this is very unlikely, but if you add an anchor to the beginning of the line, then it is even more unlikely.